pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -531,7 +531,10 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 
 	r.status.ClearDegraded()
 
-	if ls != nil {
+	// Since we don't re poll for the object we need to make sure the object wouldn't have been deleted on the patch
+	// that may have removed the finalizers.
+	// TODO We may want to just return if we remove the finalizers from the LogStorage object.
+	if ls != nil && (ls.DeletionTimestamp == nil || len(ls.GetFinalizers()) > 0) {
```

Comment:
Is the `len(ls.GetFinalizers()) > 0` needed? It seems un-needed.

Reply:
It's needed because if the deletion timestamp is set but there are still finalizers the object isn't deleted so we may want to still set a status.